browser_use/agent/playwright_script_generator.py

```diff
@@ -0,0 +1,734 @@
+        }
+        if self.browser_config.proxy:
+            args_dict["proxy"] = self.browser_config.proxy.model_dump()
+
+        # Filter out None values
+        args_dict = {k: v for k, v in args_dict.items() if v is not None}
+
+        # Format as keyword arguments string
+        args_str = ", ".join(f"{key}={repr(value)}" for key, value in args_dict.items())
+        return args_str
+
+    def _generate_context_options(self) -> str:
+        """Generates the options string for context creation based on BrowserContextConfig."""
+        if not self.context_config:
+            return "" # Default context
+
+        options_dict = {}
+
+        # Map relevant BrowserContextConfig fields to Playwright context options
+        if self.context_config.user_agent:
+            options_dict["user_agent"] = self.context_config.user_agent
+        if self.context_config.locale:
+            options_dict["locale"] = self.context_config.locale
+        if self.context_config.permissions:
+            options_dict["permissions"] = self.context_config.permissions
+        if self.context_config.geolocation:
+            options_dict["geolocation"] = self.context_config.geolocation
+        if self.context_config.timezone_id:
+            options_dict["timezone_id"] = self.context_config.timezone_id
+        if self.context_config.http_credentials:
+            options_dict["http_credentials"] = self.context_config.http_credentials
+        if self.context_config.is_mobile is not None:
+            options_dict["is_mobile"] = self.context_config.is_mobile
+        if self.context_config.has_touch is not None:
+            options_dict["has_touch"] = self.context_config.has_touch
+        if self.context_config.save_recording_path:
+             options_dict["record_video_dir"] = self.context_config.save_recording_path
+        if self.context_config.save_har_path:
+             options_dict["record_har_path"] = self.context_config.save_har_path
+
+        # Handle viewport/window size
+        if self.context_config.no_viewport:
+            options_dict["no_viewport"] = True
+        elif self.context_config.browser_window_size:
+            options_dict["viewport"] = {
+                "width": self.context_config.browser_window_size.width,
+                "height": self.context_config.browser_window_size.height
+            }
+
+        # Note: cookies_file and save_downloads_path are handled separately
+
+        # Filter out None values
+        options_dict = {k: v for k, v in options_dict.items() if v is not None}
+
+        # Format as keyword arguments string
+        options_str = ", ".join(f"{key}={repr(value)}" for key, value in options_dict.items())
+        return options_str
+
+    def _get_imports_and_helpers(self) -> List[str]:
+        """Generates necessary import statements and helper functions."""
+        # Updated _try_locate_and_act to raise PlaywrightActionError on failure
+        try_locate_and_act_code = """
```

Comment:
Move this to a separate python file so it can be type checked and linted like normal code.

You can get it as a string by importing it here and then looking at `Path(imported_module.__file__).read_text()`.